testimony/__init__.py

```diff
@@ -450,6 +462,43 @@ def bugs_report(testcases):
     )
 
 
+def tags_report(testcases):
+    """Lists the test cases matching the input tag."""
+    result = {}
+    result['tagged_testcases'] = []
+    if not SETTINGS['input_tags']:
+        print 'Input tags required for this report.  See testimony --help'
+        sys.exit()
+    # Change the input tags to lower case
+    input_tags_list = [tag.lower() for tag in SETTINGS['input_tags']]
+    for _, tests in testcases.items():
+        for testcase in tests:
+            if testcase.tags:
+                # Split the test case tags with ',' and change to lowercase
+                # and trim empty spaces
+                testcase_tags_list = [
+                    tag.lower().strip()
+                    for tag in testcase.tags.split(',')
+                ]
+                # Check if any items in either list match. If so, derive the
+                # full path of the test case. Expected sample output:
+                # `tests.test_sample.Testsample1.test_positive_login_1`
+                if (len(set(testcase_tags_list).intersection(
+                        input_tags_list)) >= 1):
+                    value = testcase.testmodule
+                    if testcase.parent_class:
+                        value = value + '.' + testcase.parent_class
+                    value = value + '.' + testcase.name
+                    value = value.replace('/', '.')
+                    value = value.replace('.py', '', 1)
+                    result['tagged_testcases'].append(value)
+    if SETTINGS['json']:
+        print json.dumps(result)
+        return
+    else:
+        print result['tagged_testcases']
```

Comment:
I designed the output primarily to get the output and directly feed into a python test suite.  But lets discuss this in detail and address in https://github.com/SatelliteQE/testimony/issues/76.